### PR TITLE
upgrade nixpkgs to 23-11

### LINKED ISSUE #18311: Upgrade nixpkgs to release `23.11`
A `nixpkgs` upgrade is available `23.11` -> https://github.com/NixOS/nixpkgs/releases/tag/23.11

This solves our requirement to upgrade `golang` beyond `1.20` 

changes form our existing 23.05 -> https://github.com/NixOS/nixpkgs/compare/23.05...23.11

### PR DESCRIPTION
**Pull request by siddarthkay**
fixes #18311

## Summary
This PR upgrades `nixpkgs` to latest release version of 23-11
ref -> https://github.com/NixOS/nixpkgs/releases/tag/23.11

## Following are the noteable upgrades : 

- `Gradle` from `8.0.1` -> `8.4`
- `Git` from `2.40.1` -> `2.42.0`
- `Curl` from `8.0.1` -> `8.4.0`
- `OpenSSL` from `3.0.8` -> `3.0.13`
- `NodeJS` from `18.16.0` -> `18.19.1`
-  `Python` from `3.10.11` -> `3.10.13`
- `Clojure` from `1.11.1.1273` -> `1.11.1.1413`
- and some more stuff

## Following derivations were removed : 
- `xcbeautify` , this was added in 23-11 -> https://github.com/NixOS/nixpkgs/pull/289446
- `idb-companion`, this was also added in 23-11 -> https://github.com/NixOS/nixpkgs/pull/296440

## Testing notes
Test Everything

## Platforms
- Android
- iOS

status: ready

### CODE REVIEW

**Review by jakubgs — APPROVED** [review]

**Review by apentori — APPROVED** [review]

**Comment by pavloburykh** [comment]
Hey @siddarthkay!

In this PR we experience issues with connection to store node. We are not reproducing this in develop, so assume this could be PR regression. See description below. 

In case this error is related to waku team please let us know. 


### ISSUE 1 Communities are not fetched, "store node is not available" in logs

**Steps:** 

1. Open any community link
2. See if community is fetched

**Actual result:** 
[Status-debug-logs - 2024-03-29T135854.407.zip](https://github.com/status-im/status-mobile/files/14804259/Status-debug-logs.-.2024-03-29T135854.407.zip)


https://github.com/status-im/status-mobile/assets/97245802/3d8bc2cd-3693-4c20-a998-eb9a97001870


log from Status log

```
03-29 13:49:32.705 32594 32694 E ReactNativeJS: 2024-03-29T11:49:32.690Z ERROR [status-im.contexts.communities.events:230] - {:message "Failed to request community info from mailserver",
03-29 13:49:32.705 32594 32694 E ReactNativeJS:  :error {:code -32000, :message "store node is not available"}}
```

log from geth log

```
DEBUG[03-29|11:49:56.025|github.com/status-im/status-go/protocol/messenger_mailserver_cycle.go:740]                                         wakuV2 storenode status verification 
INFO [03-29|11:49:56.025|github.com/status-im/status-go/protocol/messenger_mailserver_cycle.go:145]                                         Automatically switching mailserver 
INFO [03-29|11:49:56.026|github.com/status-im/status-go/protocol/messenger_mailserver_cycle.go:235]                                         Finding a new mailserver... 
INFO [03-29|11:49:56.069|github.com/status-im/status-go/protocol/messenger_mailserver_cycle.go:264]                                         connecting error                         err="lookup store-01.gc-us-central1-a.shards.test.statusim.net on [::1]:53: read udp [::1]:42714->[::1]:53: read: connection refused"
INFO [03-29|11:49:56.069|github.com/status-im/status-go/protocol/messenger_mailserver_cycle.go:264]                                         connecting error                         err="lookup store-02.gc-us-central1-a.shards.test.statusim.net on [::1]:53: read udp [::1]:48553->[::1]:53: read: connection refused"
INFO [03-29|11:49:56.069|github.com/status-im/status-go/protocol/messenger_mailserver_cycle.go:264]                                         connecting error                         err="lookup store-01.ac-cn-hongkong-c.shards.test.statusim.net on [::1]:53: read udp [::1]:46830->[::1]:53: read: connection refused"
INFO [03-29|11:49:56.069|github.com/status-im/status-go/protocol/messenger_mailserver_cycle.go:264]                                         connecting error                         err="lookup store-02.ac-cn-hongkong-c.shards.test.statusim.net on [::1]:53: read udp [::1]:40038->[::1]:53: read: connection refused"
INFO [03-29|11:49:56.069|github.com/status-im/status-go/protocol/messenger_mailserver_cycle.go:264]                                         connecting error                         err="lookup store-01.do-ams3.shards.test.statusim.net on [::1]:53: read udp [::1]:33120->[::1]:53: read: connection refused"
INFO [03-29|11:49:56.069|github.com/status-im/status-go/protocol/messenger_mailserver_cycle.go:264]                                         connecting error                         err="lookup store-02.do-ams3.shards.test.statusim.net on [::1]:53: read udp [::1]:54447->[::1]:53: read: connection refused"
WARN [03-29|11:49:56.069|github.com/status-im/status-go/protocol/messenger_mailserver_cycle.go:271]                                         No mailservers available 
```

**Comment by mariia-skrypnyk** [comment]
Hi @siddarthkay !

Thanks for your PR.

Catched a problem that firstly was checked on nightly and another rebased PR but both of them have no such befaviour.
Also I asked to check another QA and the behaviour is similar.
So.

ISSUE 1: User stuсks on Fetching Community screen and can not get forward to the Community screen to join it

1. Login to app as User A
2. Send community link as User B to User A
3. Tap on link as User A

Expected result: User A sees white screen with "Fetching Community" text 1-2 sec and then is redirected to the Community screen to join it

Actual result: user stucks on white screen with "Fetching Community" text

Please, look use attached logs if needed:
[logs_User A.zip](https://github.com/status-im/status-mobile/files/14804296/logs_User.A.zip)
[logs_User B.zip](https://github.com/status-im/status-mobile/files/14804297/logs_User.B.zip)

**Comment by siddarthkay** [comment]
Thanks for testing @pavloburykh : I'll take a look.
On the surface it does look network related (possible waku issue, not sure yet) 
but I'll try to reproduce it first and get back to you!

File: nix/overlay.nix [hunk 1]
```diff
@@ -60,15 +60,21 @@ in {
     version = "15.0";
     allowHigher = true;
   };
-  go = super.go_1_19;
+  go = super.go_1_20;
   clang = super.clang_15;
-  buildGoPackage = super.buildGo119Package;
-  buildGoModule = super.buildGo119Module;
+  buildGoPackage = super.buildGo120Package;
+  buildGoModule = super.buildGo120Module;
```
Comment (siddarthkay) [inline nix/overlay.nix]: I think i should revert the go upgrade here to 1_20 and do that separately.
Maybe that causes issues

**Comment by mariia-skrypnyk** [comment]
Hi @siddarthkay !

Also found extra scenario caused by connection to store node. May be useful to reproducing problem.

**ISSUE 2: User created by using seed phrase has no restored data**

1. Install app
2. I'm new -> Use recovery phrase
3. Go trough onboarding
4. Wait 5 min till data appears

**Expected result:** user will be recovered with data
**Actual result:** no restored data

**Comment by mariia-skrypnyk** [comment]
@siddarthkay I've checked new builds but unfortunately all issues are still exist on both platforms:

iOS/Android

1. communities are not fetched
2. user is not recovered using a seed phrase

**Review by jo-mut — APPROVED** [review]

File: nix/overlay.nix (hunk 1 above)
Comment (jakubgs) [reply nix/overlay.nix]: I agree.

Comment (siddarthkay) [reply nix/overlay.nix]: reverting go to 1.19 and getting rid of the `netgo` build flag for go-mobile fixed the issue

**Comment by siddarthkay** [comment]
@pavloburykh @mariia-skrypnyk : 
Sorry for the ping on the weekend.
I checked and with the latest commit I was able to fetch community properly.
Please check when you get a chance.

**Comment by pavloburykh** [comment]
@siddarthkay I am not reproducing [ISSUE 1](https://github.com/status-im/status-mobile/pull/19369#issuecomment-2027189276) anymore. 

PR is ready for merge. 

> I checked and with the latest commit I was able to fetch community properly.

@siddarthkay by last commit you mean these changes https://github.com/status-im/status-mobile/pull/19369#discussion_r1544538783 ? If yes - then it is weird, because issue actually was reproducible for me and you after this commit when we checked on Friday. 

Or you committed some other changes which fixed the bug?

**Comment by siddarthkay** [comment]
Thanks for testing @pavloburykh ! 
The issue was actually fixed on Saturday. 
The root cause was `netgo` build flag and golang bump to `1.20`
Reverting those changes fixed the issue.

**Comment by pavloburykh** [comment]
> Thanks for testing @pavloburykh ! The issue was actually fixed on Saturday. The root cause was `netgo` build flag and golang bump to `1.20` Reverting those changes fixed the issue.

Great, thank you @siddarthkay!

**Pull request merged**